cpp/src/arrow/compute/kernels/api.h

```diff
@@ -0,0 +1,31 @@
+// Licensed to the Apache Software Foundation (ASF) under one
+// or more contributor license agreements.  See the NOTICE file
+// distributed with this work for additional information
+// regarding copyright ownership.  The ASF licenses this file
+// to you under the Apache License, Version 2.0 (the
+// "License"); you may not use this file except in compliance
+// with the License.  You may obtain a copy of the License at
+//
+//   http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing,
+// software distributed under the License is distributed on an
+// "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+// KIND, either express or implied.  See the License for the
+// specific language governing permissions and limitations
+// under the License.
+
+#pragma once
+
+#include "arrow/compute/visibility.h"
+#include "arrow/status.h"
+
+namespace arrow::compute {
+
+/// \brief Initialize the compute module.
+///
+/// Register the compute kernel functions to be available on the
+/// global FunctionRegistry.
+ARROW_COMPUTE_EXPORT Status Initialize();
```

Comment:
Will it be better to move this declaration into `compute/api.h`? It feels like this "top-level" compute function should be placed at the "top-level" directory of compute module.

Reply:
This is exactly as I suggested. Thanks for the updates.